pulp_rpm/app/tasks/publishing.py

```diff
@@ -254,31 +256,29 @@ def get_checksum_type(name, checksum_types):
     if metadata:
         checksum_type = metadata
 
-    return getattr(cr, checksum_type.upper(), cr.SHA256)
+    return getattr(cr, getattr(CHECKSUM_TYPES, checksum_type.upper()), cr.SHA256)
 
 
 def publish(
     repository_version_pk,
-    gpgcheck_options=None,
-    metadata_signing_service=None,
-    checksum_types=None,
-    sqlite_metadata=False,
+    publish_settings_pk,
```

Comment:
Ugh, I might have to rethink how to structure this to accomodate the migration plugin. Which uses publish() directly.